src/liballoc/arc.rs

```diff
@@ -1211,8 +1360,49 @@ impl<T> From<T> for Arc<T> {
     }
 }
 
+#[unstable(feature = "shared_from_slice", issue = "40475")]
+impl<'a, T: Clone> From<&'a [T]> for Arc<[T]> {
+    #[inline]
+    default fn from(v: &[T]) -> Arc<[T]> {
```

Comment:
We currently try to avoid exposing the usage of specialization right now, mind changing this to use a local `ArcFromSlice` trait which can be locally specialized?